balboa-agent/src/main/scala/HttpMetricQueue.scala

```diff
@@ -0,0 +1,60 @@
+package com.socrata.balboa.agent
+
+import java.net.URL
+import java.util.Date
+
+import com.socrata.balboa.metrics.{EntityJSON, Metric, MetricJSON}
+import com.socrata.metrics.{IdParts, MetricQueue}
+import com.stackmob.newman.ApacheHttpClient
+import com.stackmob.newman.dsl.POST
+import com.stackmob.newman.response.HttpResponseCode.Ok
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.json4s.{DefaultFormats, Extraction, Formats}
+import org.json4s.jackson.JsonMethods.{pretty, render}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class HttpMetricQueue(balboaHttpURL: String, timeout: Duration) extends MetricQueue with StrictLogging {
+
+  implicit val httpClient = new ApacheHttpClient
+  implicit val jsonFormats: Formats = DefaultFormats
+  implicit val executionContext = ExecutionContext.global
+
+  /**
+   * Interface for receiving a Metric
+   *
+   * @param entity Entity which this Metric belongs to (ex: a domain).
+   * @param name Name of the Metric to store.
+   * @param value Numeric value of this metric.
+   * @param timestamp Time when this metric was created.
+   * @param recordType Type of metric to add, See [[Metric.RecordType]] for more information.
+   */
+  override def create(entity: IdParts,
+             name: IdParts,
+             value: Long,
+             timestamp: Long = new Date().getTime,
+             recordType: Metric.RecordType = Metric.RecordType.AGGREGATE): Unit = {
+
+    val metricToWrite = EntityJSON(timestamp, Map(name.toString -> MetricJSON(value, recordType.toString)))
+
+    val url = new URL(s"$balboaHttpURL/metrics/$entity")
+
+    val request = POST(url).setBody(pretty(render(Extraction.decompose(metricToWrite)))).apply
+
+    val requestWithTimeout = Future { Await.result(request, timeout) }
+
+    requestWithTimeout.onComplete {
```

Comment:
I think the approach to take here is to just keep looping, infinitely and retrying when the send fails. Failed requests will wait, then retry. You can do a little exponential backoff up to a maximum wait time of lets say 2 minutes (stored in the config file) and then keep retrying every 2 minutes.

It's naive, and we may have to tune it a little bit before releasing but my reasoning (such as it is) goes thusly.

The previous behavior was infinite loop on a failed send. If we change that behavior (and propagate exceptions), we will have to change behavior up in the call stack, but the ActiveMQ connection would still behave the same. Probably more work than we need at this time.

Which reminds me, we should be logging improperly formatted metric save requests. Not sure what's happening right now. I added a Jira for you to confirm/add the behavior.